torchrl/objectives/ppo.py

```diff
@@ -434,11 +434,14 @@ def forward(self, tensordict: TensorDictBase) -> TensorDictBase:
             ess = (2 * lw.logsumexp(0) - (2 * lw).logsumexp(0)).exp()
             batch = log_weight.shape[0]
 
-        if not advantage.shape == log_weight.shape:
-            raise RuntimeError(
-                f"advantage.shape and log_weight.shape do not match (got {advantage.shape} "
-                f"and {log_weight.shape})"
-            )
+        if advantage.shape != log_weight.shape:
+            try:
+                log_weight = log_weight.view_as(advantage)
+            except Exception as err:
+                raise RuntimeError(
+                    f"advantage.shape and log_weight.shape do not match (got {advantage.shape} "
+                    f"and {log_weight.shape})"
+                ) from err
```

Comment:
I'm not sure we want to keep these changes. Let me explain what usually happens:

We usually have an input tensor of observations and actions of shape `[*B, F]` and `[*B, A]` respectively.
We pass the observation through the model, get a distribution with params `[*B, P]`. Then we measure the probability of the action. This will give a tensor of shape `[*B]` because the last dim vanishes in `log_prob` (the probability of a multivariate event is still a scalar, so to say).
Hence in PPO we unsqueeze the last dimension to get `[*B, 1]` which is also the expected size of the reward, done state and value.

For some reason, after my refactoring I ended up with a log_prob of size `[*B, 1]`, hence the unsqueeze was producing a shape that was unwanted. But this may be a bug on my side